#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.06959747313523405 0.059059810401071854 -0.995825351418902 1.738326892701972
  position 0.09305439870090047 0.5490302299165757 0.13829315178599028
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0.41 1.04 0
  floorSize 5 5
  floorAppearance Parquetry {
    type "dark strip"
  }
  wallHeight 0.2
}
Robot {
  translation -0.860023 0 0.06
  rotation 0 1 0 1.57
  children [
    DEF BODY_SHAPE Shape {
      appearance PBRAppearance {
        baseColor 0.666667 0 0.498039
        roughness 1
        metalness 0
      }
      geometry Capsule {
        height 0.2
        radius 0.05
      }
    }
    DEF PITCH_WHEEL HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 1
      }
      device [
        PositionSensor {
          name "p_sensor"
        }
        DEF pitch_motor RotationalMotor {
          name "pitch_motor"
        }
      ]
      endPoint Solid {
        children [
          DEF pitch Shape {
            appearance PBRAppearance {
              roughness 1
            }
            geometry Cylinder {
              height 0.03
              radius 0.06
              subdivision 30
            }
          }
        ]
        name "solid(2)"
        boundingObject USE pitch
        physics Physics {
        }
      }
    }
    DEF THIGHS Solid {
      translation -0.02 4.73704e-08 8.20112e-08
      rotation 0.5773509358560258 -0.577349935856137 0.577349935856137 2.09439
      children [
        DEF LEGS Shape {
          appearance PBRAppearance {
          }
          geometry DEF leg_shape Capsule {
            height 0.2
            radius 0.005
            subdivision 24
          }
        }
      ]
      name "thighs"
      boundingObject USE LEGS
      physics Physics {
      }
    }
    DEF TOES Solid {
      translation -0.008 0.11 -0.000492821
      rotation -0.35740612701840735 0.8628563066501258 -0.3574071270187627 -1.57
      children [
        DEF TOES Shape {
          appearance PBRAppearance {
          }
          geometry DEF toe_shape Capsule {
            height 0.03
            radius 0.005
            subdivision 24
          }
        }
      ]
      name "thighs(1)"
      boundingObject USE TOES
      physics Physics {
      }
    }
    DEF TOES Solid {
      translation -0.008 -0.11 -0.00038
      rotation 0.43792685970818224 0.8305777339207279 0.3440358897865722 -1.649255307179586
      children [
        DEF TOES Shape {
          appearance PBRAppearance {
          }
          geometry DEF toe_shape Capsule {
            height 0.03
            radius 0.005
            subdivision 24
          }
        }
      ]
      name "thighs(2)"
      boundingObject USE TOES
      physics Physics {
      }
    }
  ]
  name "chad"
  boundingObject USE BODY_SHAPE
  physics Physics {
  }
  controller "chad_controller"
}
Robot {
  translation 0 -5.0981e-08 0.05
  rotation 1 0 0 -1.5707953071795862
  children [
    Hinge2Joint {
      jointParameters HingeJointParameters {
        axis 0 1 0
      }
      jointParameters2 JointParameters {
      }
      device [
        DEF yaw_brake Brake {
        }
        DEF yaw_motor RotationalMotor {
          name "yaw_motor"
        }
        DEF yaw_sensor PositionSensor {
          name "yaw_sensor"
        }
      ]
      device2 [
        DEF roll_break Brake {
        }
        DEF roll_motor RotationalMotor {
          name "roll_motor"
        }
        DEF roll_sensor PositionSensor {
          name "roll_sensor"
        }
      ]
      endPoint Solid {
        rotation 0 1 0 0
        children [
          DEF hinge_shape Shape {
            appearance PBRAppearance {
              baseColor 0.666667 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.12
              radius 0.065
            }
          }
        ]
        boundingObject USE hinge_shape
        physics Physics {
        }
      }
    }
    USE BODY_SHAPE
    DEF legs_2 Solid {
      translation 3.85653e-08 -0.04 9.01056e-09
      rotation 5.94698e-07 -1 -3.69431e-07 -1.5708053071795867
      children [
        USE LEGS
      ]
      name "solid(1)"
      boundingObject USE LEGS
      physics Physics {
      }
    }
    DEF toes_2 Solid {
      translation -0.105858 -0.0302843 3.4e-05
      rotation 0.35740699927911007 0.8628559982596194 0.35740699927911007 -1.7177753071795863
      children [
        USE TOES
      ]
      name "solid(2)"
      boundingObject USE TOES
      physics Physics {
      }
    }
    DEF toes_2_2 Solid {
      translation 0.113 -0.03 3.45153e-05
      rotation -0.3574030741036037 0.8628571789039632 -0.35740807410464037 -1.7177753071795863
      children [
        USE TOES
      ]
      name "solid(3)"
      boundingObject USE TOES
      physics Physics {
      }
    }
  ]
  name "alexa"
  boundingObject USE BODY_SHAPE
  physics Physics {
  }
  controller "alexa_controller"
}
